### PR TITLE
Check if event handlers are supported and fix refactor error

### LINKED ISSUE #160: Frirefox show false negatives on the single resolution webcam test
On camresolutionstest.js on the collectAndAnalyzeStats_ function calling addEventListener on the videoTrack throws an exception since the videoTrack doesn't contain this method. This result in a false negative since the method execution is interrupted as this exception is not catched anywhere.

Tested on Windows Firefox 36.0.1, 39.0 and 42.0 and Mac 42.0

### PR DESCRIPTION
**Pull request by KaptenJansson**
Fixes #160

### CODE REVIEW

File: src/js/camresolutionstest.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ CamResolutionsTest.prototype = {
             ' not supported');
         this.maybeContinueGetUserMedia();
```
Comment (andresusanopinto) [inline src/js/camresolutionstest.js]: Can you move this.maybeContinueGetUserMedia out of the if and remove the test.done() ?

I wanted the code flow here to be simplified as in "maybeContinueGetUserMedia" is the only function to call test.done().

Comment (KaptenJansson) [inline src/js/camresolutionstest.js]: Done.

File: src/js/camresolutionstest.js [hunk 2]
```diff
@@ -83,15 +83,15 @@ CamResolutionsTest.prototype = {
...
-        this.test.reportError('getUserMedia failed with error: ' + error.name);
-        this.test.done();
+        this.test.reportError('getUserMedia failed with error: ' + error);
       }
+      this.maybeContinueGetUserMedia();
     }.bind(this));
   },
 
   maybeContinueGetUserMedia: function() {
+    console.log(this.currentResolution, this.resolutions.length);
```
Comment (andresusanopinto) [inline src/js/camresolutionstest.js]: remove debug msg

**Comment by andresusanopinto** [comment]
lgtm, but remove the debug message

File: src/js/camresolutionstest.js (hunk 2 above)
Comment (KaptenJansson) [inline src/js/camresolutionstest.js]: Doh, sloppy of me. Done.

**Comment by KaptenJansson** [comment]
Hmm this now has the side effect of checking for video frames even though getUserMedia has failed ;)

**Pull request merged**